src/node.ts

```diff
@@ -0,0 +1,396 @@
+// Copyright (c) 2018 Zilliqa
+// This source code is being disclosed to you solely for the purpose of your participation in
+// testing Zilliqa. You may view, compile and run the code for that purpose and pursuant to
+// the protocols and algorithms that are programmed into, and intended by, the code. You may
+// not do anything else with the code without express permission from Zilliqa Research Pte. Ltd.,
+// including modifying or publishing the code (or any part of it), and developing or forming
+// another public or private blockchain network. This source code is provided ‘as is’ and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose
+// and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+import { validateArgs } from './util';
+import * as util from './util';
+
+type callback = (error: any, data: any) => any;
+
+const utils = {
+  generatePrivateKey: util.generatePrivateKey,
+  verifyPrivateKey: util.verifyPrivateKey,
+  getAddressFromPrivateKey: util.getAddressFromPrivateKey,
+  getPubKeyFromPrivateKey: util.getPubKeyFromPrivateKey,
+  createTransactionJson: util.createTransactionJson,
+};
+
+export default class ZNode {
+  url: string;
+  apiUrl: string;
+  utils: typeof utils;
+
+  constructor(args: any) {
+    validateArgs(args, {
+      url: [util.isUrl],
+    });
+
+    this.url = args.url;
+    this.apiUrl = 'https://api.zilliqa.com';
+    this.utils = utils;
+  }
+
+  /**
+   * isConnected
+   *
+   * returns network id
+   *
+   * @param {callback} cb
+   */
+  isConnected = (cb: callback) => {
```

Comment:
since this is a class, i think the syntax is a little confusing. It should be:
```
isConnected(cb: callback) {...}
```

I don't see any advantage using arrow function here.

Reply:
I have the habit of doing this even for class methods, because arrow functions save me the headache of dealing with `this` binding. Not a big issue, though. I generally just prefer this syntax.